pkg/operation/botanist/component/kubecontrollermanager/kube_controller_manager.go

```diff
@@ -405,7 +405,7 @@ func (k *kubeControllerManager) Deploy(ctx context.Context) error {
 	}
 
 	// TODO(rfranzke): Remove in a future release.
-	return kutil.DeleteObject(ctx, k.seedClient, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "kube-controller-manager", Namespace: k.namespace}})
+	return kutil.DeleteObject(ctx, k.seedClient, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "kube-controller-manager-server", Namespace: k.namespace}})
```

Comment:
From the code it looks like the secret `kube-controller-manager-server` is first created, and then deleted, is this intended? If yes, why not use the `secretNameServer` constant here to make it more explicit, and perhaps leave a comment why all of this is correct? The comment "remove in a future release" suggests that this is temporary, could you add a short explanation why?
Note: from testing I got what's happening, you are replacing the original secret with a secret that is named e.g. `kube-controller-manager-server-dce8c938`, but this is not really obvious in the code.

Reply:
The secrets manager will use `kube-controller-manager-server` as prefix for the secret name and will add some hash suffix to it. Deleting it here is just to cleanup the old secret whose name was precisely `kube-controller-manager-server`. 

> The comment "remove in a future release" suggests that this is temporary, could you add a short explanation why?

After we have cleaned up all old secrets we can delete this code again.

I would prefer to keep it as is for now since it's already done similarly at many other places and adapting this brings little improvement given that this deletion is only temporary anyways. OK?

Reply:
OK, makes sense.